jfr-streaming/src/g1GcTest/java/io/opentelemetry/contrib/jfr/metrics/AbstractMetricsTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jfr.metrics;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.testing.assertj.MetricAssert;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader;
+import java.util.Collection;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.BeforeAll;
+
+public class AbstractMetricsTest {
+
+  static SdkMeterProvider meterProvider;
+  static InMemoryMetricReader metricReader;
+  static boolean isInitialized = false;
+
+  @BeforeAll
+  static void initializeOpenTelemetry() {
+    if (isInitialized) {
+      return;
+    }
+    isInitialized = true;
+    metricReader = InMemoryMetricReader.create();
+    meterProvider = SdkMeterProvider.builder().registerMetricReader(metricReader).build();
+    GlobalOpenTelemetry.set(OpenTelemetrySdk.builder().setMeterProvider(meterProvider).build());
+    JfrMetrics.enable(meterProvider);
```

Comment:
Yup I agree with all those points. 
>Right now that's required because JfrMetrics can't be cancelled once started
Maybe `@BeforeEach` test I can reinitialize JFR streaming and all openTelemetry variables, and `@AfterEach` test the JFR streaming can be ended.  I think that should be enough for tests to fully clean up after themselves.

 > Should be put in a common place where it can be used by each of the different testing suites

I was trying to figure out a good way of doing this today. Maybe something related to test fixtures. I'll have to investigate more.

Reply:
Update:
I renamed the base test class to `AbstractJfrTest`. I was able to get the `testFixtures` working. So now the class resides in a single place, instead of being duplicated for each GC type.